### ISSUE TITLE
MissingIDFieldWarning Warning

### ISSUE THREAD

**Issue by bdsomer** [ucbds-infra/otter-grader#677]
**Describe the bug**
When using `otter assign` on a notebook whose code cells have the `id` field, I get the following warning:

```
.../otter/assign/notebook_transformer.py:386: MissingIDFieldWarning: Code cell is missing an id field, this will become a hard error in future nbformat versions. You may want to use `normalize()` on your notebooks before validations (available since nbformat 5.1.4). Previous versions of nbformat are fixing this issue transparently, and will stop doing so in the future.
  _, nb = normalize(nb)
```

**To Reproduce**
Steps to reproduce the behavior: Use `otter assign` on a notebook whose code cells have the `id` field.

**Expected behavior**
There is no warning when using `otter assign` on properly formatted notebooks that have the `id` field.

**Versions**
Python 3.11.3, Otter 5.0.1

**Additional context**
It seems that this is caused by https://github.com/ucbds-infra/otter-grader/blob/8001f6f1f4150124cdb4e14865f6f9fc717fd75c/otter/assign/notebook_transformer.py#L129 which was added to fix #340 (as the comment describes).

Also I am not sure if this issue is covered by #301; if that is the case, feel free to close this issue.

**Comment by chrispyles** [comment]
What version of nbformat do you have installed? Also can you send a notebook that reproduces this? I thought this was fixed by [this](https://github.com/ucbds-infra/otter-grader/blob/master/otter/assign/notebook_transformer.py#L385), which was added in v5.0.0 (cf. #658), but apparently not.

**Comment by taylorgibson** [comment]
I'm seeing a similar issue:

```
/home/gibson/mambaforge/envs/dstraining/lib/python3.11/site-packages/otter/assign/notebook_transformer.py:385: MissingIDFieldWarning: Code cell is missing an id field, this will become a hard error in future nbformat versions. You may want to use `normalize()` on your notebooks before validations (available since nbformat 5.1.4). Previous versions of nbformat are fixing this issue transparently, and will stop doing so in the future.
  _, nb = normalize(nb)
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
```

Attaching a notebook file that produces this output.
[demo.zip](https://github.com/ucbds-infra/otter-grader/files/12195735/demo.zip)

Python version: 3.11.4
Otter-Grader version: 5.0.2

**Comment by chrispyles** [comment]
@taylorgibson what version of nbformat do you have installed?

**Comment by taylorgibson** [comment]
`nbformat` is version 5.9.1

**Comment by taylorgibson** [comment]
Here's the full list of packages in the environment in case that's helpful:
[myenv.txt](https://github.com/ucbds-infra/otter-grader/files/12207319/myenv.txt)

**Comment by chrispyles** [comment]
This should be fixed in v5.1.0, where I've removed the patch that strips the cell IDs.
